lib/ash/actions/create/create.ex

```diff
@@ -111,6 +111,15 @@ defmodule Ash.Actions.Create do
         opts[:upsert_identity] || get_in(changeset.context, [:private, :upsert_identity])
       end
 
+    upsert_fields =
+      if opts[:upsert?] do
```

Comment:
I don't think we need the conditional do we? Should just be able to always use:

```elixir
opts[:upsert_fields] || get_in(changeset.context, [:private, :upsert_fields]) ||
  action.upsert_fields
```

Reply:
ok, seems reasonable. I changed it, tests still pass. (will push)